server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java

```diff
@@ -100,19 +101,23 @@ public PermissionTemplate createPermissionTemplate(String name, @Nullable String
   }
 
   public void updatePermissionTemplate(Long templateId, String newName, @Nullable String newDescription, @Nullable String newKeyPattern) {
-    PermissionTemplateUpdater.checkSystemAdminUser(userSession);
+    checkGlobalAdminUser(userSession);
     validateTemplateName(templateId, newName);
     validateKeyPattern(newKeyPattern);
     permissionTemplateDao.updatePermissionTemplate(templateId, newName, newDescription, newKeyPattern);
   }
 
   public void deletePermissionTemplate(Long templateId) {
-    PermissionTemplateUpdater.checkSystemAdminUser(userSession);
+    checkGlobalAdminUser(userSession);
     permissionTemplateDao.deletePermissionTemplate(templateId);
   }
 
+  /**
+   * @deprecated since 5.2 can be removed when Ruby doesn't rely on PermissionTemplateService
+   */
+  @Deprecated
   public void addUserPermission(String templateKey, String permission, String userLogin) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)